astropy/io/fits/diff.py

```diff
@@ -1112,7 +1112,8 @@ def _report(self):
             return
 
         for index, values in self.diff_pixels:
-            index = [x + 1 for x in reversed(index)]
+            # Convert to int to avoid np.int64 in list repr.
+            index = [int(x + 1) for x in reversed(index)]
```

Comment:
this bit is cherry-picked from https://github.com/astropy/astropy/pull/15065

Reply:
In that case, I pinged mhvk for review. Thanks!